### ISSUE TITLE
`error-stack` UI tests fail under `nightly-2022-07-20`

### ISSUE THREAD

**Issue by indietyp** [hashintel/hash#897]
nightly-2022-07-20 (rust 1.64 stable) introduced https://github.com/rust-lang/rust/pull/98320, which means that the UI and acceptance tests for the macro errors fail on toolchains newer than that.

https://github.com/hashintel/hash/blob/45598cd1b39d63fbce1614ebbab799b19ee60173/packages/libs/error-stack/tests/ui/macro_invalid_args.stderr#L49

This is changed to:

```
= note: this error originates in the macro `$crate::report` which comes from the expansion of the macro `bail` (in Nightly builds, run with -Z macro-backtrace for more info)
```
---

https://github.com/hashintel/hash/blob/45598cd1b39d63fbce1614ebbab799b19ee60173/packages/libs/error-stack/tests/ui/macro_invalid_args.stderr#L62

This is changed to:

```
= note: this error originates in the macro `$crate::report` which comes from the expansion of the macro `ensure` (in Nightly builds, run with -Z macro-backtrace for more info)
```

---

The ``which comes from the expansion of the macro `...` `` part is new.

**Comment by indietyp** [comment]
To fix this, we could either

1) update the tests and CI (single PR) - because we already only compiletest on nightly
2) use `rustversion` to conditionally emit additional code on toolchains that are newer than `2022-07-20`

(from the logs of #892 it also seems like the code to fetch the nightly toolchain version seems broken)

@TimDiekmann what do you think?

**Comment by TimDiekmann** [comment]
Thanks for filing this!

I think the easiest way is to simply bump the nightly version used in CI, this will also fix #898. I will do this tomorrow.

> (from the logs of https://github.com/hashintel/hash/pull/892 it also seems like the code to fetch the nightly toolchain version seems broken)

Currently, it uses the first `rust-toolchain.toml` file it finds in the repository to determine the nightly version, this is why we currently try to keep the nightly versions on par between the different projects (`package/engine`, `packages/libs/error-stack` and `packages/graph`(on `dev/graph`) ). Probably there is a better way to this, but I didn't investigated this yet.

**Comment by indietyp** [comment]
What I specifically mean is the error:

```
Run nightly=$(cat $(find . -name rust-toolchain.toml) | grep channel | cut -d\" -f2)
parse error: Invalid literal at line 2, column 0
use toolchains: 
```

Which turns up in the CI.
